### PR TITLE
Add functions to manager folders and files

### PR DESCRIPTION
**Pull request by wevnasc**
I started to change the code structure to steps as a pipeline as I sad on #1, Also I add some tests, I think that could be cool add some kind of CI to execute them before the merge.

I also changed the folder where the logs are created, what was happening for me was that the config folder was created after the log, so it throw some hided erros. Now I think that should work because `tmp` directory is already there aways for linux and mac at least.

### CODE REVIEW

File: backgroundchanger/__main__.py [hunk 1]
```diff
@@ -42,7 +50,15 @@ def do_wal(photo, do_colors=True):
     utils.reload_gala()
 
 def main():
-    parsed = setup()
+    try:
+        start_log()
+        create_config_folders()
+        create_config_file()
+    except Exception as error:
+        logging.critical(error.message)
```
Comment (conruggles) [inline backgroundchanger/__main__.py]: Can you log the stack trace as well? Thanks!

**Comment by conruggles** [comment]
Yeah I like your changes, I did leave one comment. 

I think it would be a good idea to run tests as a merge check, but I think some should be added for the main code as well before adding that pipeline in. I don't expect you to do that, but just letting you know.

Thank you for doing so much work on improving this project, I really appreciate it.

**Comment by wevnasc** [comment]
sorry @rugglcon, but I didn't see your comment, you put that in my changes?

**Review by conruggles — CHANGES_REQUESTED** [review]

**Comment by conruggles** [comment]
@wevnasc ha oops, forgot to click "Submit review". It should be there now.

File: backgroundchanger/__main__.py (hunk 1 above)
Comment (wevnasc) [reply backgroundchanger/__main__.py:59]: I am not sure how can I do that, but I think that should be that, right?

File: backgroundchanger/__main__.py [hunk 2]
```diff
@@ -42,7 +50,16 @@ def do_wal(photo, do_colors=True):
     utils.reload_gala()
 
 def main():
-    parsed = setup()
+    try:
+        start_log()
+        create_config_folders()
+        create_config_file()
+    except Exception as error:
+        logging.critical(error)
```
Comment (wevnasc) [inline backgroundchanger/__main__.py]: @rugglcon I am not sure how can I do that, but I think that should be that, right?

Comment (conruggles) [reply backgroundchanger/__main__.py]: You should be able to do `logging.exception(error)`, and that will log the entire `Exception` object, including the message and stack trace.

File: requirements-dev.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+pytest==6.1.1
```
Comment (conruggles) [inline requirements-dev.txt]: Probably don't need to pin the version down, if something breaks I should be able to respond quickly enough to make sure the build passes again. Personally, I don't really like pinned versions.

**Review by conruggles — CHANGES_REQUESTED** [review]

File: requirements-dev.txt (hunk 3 above)
Comment (wevnasc) [reply requirements-dev.txt]: I changed to get always the latest one ✌️

File: backgroundchanger/__main__.py (hunk 2 above)
Comment (wevnasc) [reply backgroundchanger/__main__.py]: I see, thanks!

File: requirements-dev.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+pytest==*
```
Comment (conruggles) [inline requirements-dev.txt]: You can just remove completely the `==*`
```suggestion
pytest
```

File: requirements.txt [hunk 5]
```diff
@@ -1,3 +1,3 @@
-requests==2.24.0
-pywal==3.3.0
-wheel==0.35.1
\ No newline at end of file
+requests==*
```
Comment (conruggles) [inline requirements.txt]: Same thing here, can just remove the `==*` from these lines as well.

**Review by conruggles — CHANGES_REQUESTED** [review]

**Review by conruggles — APPROVED** [review]
Looks great! thanks so much

**Pull request merged**